runtime/debugger.c

```diff
@@ -139,7 +140,10 @@ static void open_connection(void)
   if (dbg_socket == -1)
     caml_fatal_error("_open_osfhandle failed");
 #endif
-  dbg_in = caml_open_descriptor_in(dbg_socket);
+  dbg_socket_in = dup(dbg_socket);
```

Comment:
I have to use `_dup`, right?

Reply:
To be honest, I have no idea.

If we want to actually call `dup` on unix, then the best is probably to ask @dra27 about what is the right thing to do here.

But see #9786: the original implementation corresponds to a common pattern in OCaml programming with the Unix module, which, I think, should not be considered a bug. So, I think I would prefer keeping the initial implementation, bug guard somewhere else about the double close of file descriptors used in channels.

Reply:
`dup` is aliased to `_dup`, so it'll work - it would be better to use `_dup` on Windows, but goodness knows if/when Microsoft will actually remove the POSIX aliases.

What you're doing should work fine on Windows; I have no comment on whether it's necessarily what's wanted on either platform.